### PR TITLE
Add vision requirement to image transforms

### LINKED ISSUE #20627: When Pillow is not installed, importing from transformers.image_transforms raises an unclear NameError
### System Info

- `transformers` version: 4.25.1
- Platform: macOS-11.6.8-x86_64-i386-64bit
- Python version: 3.10.8
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): not installed (NA)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@amyeroberts @NielsRogge

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

1. Create and activate clean Python environment 
   ```sh
   python3 -m venv venv
   source venv/bin/activate
   ```
2. Install `transformers` and its direct dependencies
   ```sh
   pip install transformers
   ```
3. Attempt to import `transformers.image_transforms` or [one of its publicly-documented members](https://huggingface.co/docs/transformers/internal/image_processing_utils#transformers.image_transforms.center_crop)
   ```sh
   python -c 'from transformers.image_transforms import center_crop'
   ```
4. Encounter a `NameError`
   ```
   None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
   Traceback (most recent call last):
     File "<string>", line 1, in <module>
     File "/Users/miliu/venv/lib/python3.10/site-packages/transformers/image_transforms.py", line 51, in <module>
       def to_channel_dimension_format(image: np.ndarray, channel_dim: Union[ChannelDimension, str]) -> np.ndarray:
   NameError: name 'ChannelDimension' is not defined
   ```

### Expected behavior

Rather than a `NameError` on import (caused by [`to_channel_dimension_format()`'s signature type annotation containing `ChannelDimension`](https://github.com/huggingface/transformers/blob/7586a1a367f5974e099e1be2fa8a751aa766179f/src/transformers/image_transforms.py#L51), which is [conditionally imported](https://github.com/huggingface/transformers/blob/7586a1a367f5974e099e1be2fa8a751aa766179f/src/transformers/image_transforms.py#L29) only [when `PIL` is available](https://github.com/huggingface/transformers/blob/7586a1a367f5974e099e1be2fa8a751aa766179f/src/transformers/utils/import_utils.py#L566-L567)), I would've expected something like `transformers.rescale`'s user experience, where it helpfully recommends installing `Pillow` when one attempts to use it:

```python
>>> from transformers import rescale
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
>>> rescale()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/miliu/venv/lib/python3.10/site-packages/transformers/utils/dummy_vision_objects.py", line 14, in rescale
    requires_backends(rescale, ["vision"])
  File "/Users/miliu/venv/lib/python3.10/site-packages/transformers/utils/import_utils.py", line 997, in requires_backends
    raise ImportError("".join(failed))
ImportError: 
rescale requires the PIL library but it was not found in your environment. You can install it with pip:
`pip install pillow`. Please note that you may need to restart your runtime after installation.
```

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Addresses issue when importing functions in the `image_transforms` module if Pillow isn't installed. Namely: 
* functions which don't require Pillow have errors when importing. This was because other objects were only imported if Pillow was available e.g. `ChannelDimension`.  Resolved by rearranging imports. 
* Unuseful error message hiding that the issue is with Pillow in the environment. Resolved by adding a `vision_required` decorator to allow for errors only on select functions.  

Fixes #20627 

This needs to be merged in before the transforms can be removed from the init in #20704

Snippet below shows new behaviour when running in an environment without Pillow installed: 

```
>>> import numpy as np
>>> 
>>> # Show we can import both methods without issue
>>> from transformers.image_transforms import rescale
>>> from transformers.image_transforms import center_crop
>>> 
>>> img = np.random.randint(0, 256, (3, 244, 360))
>>> 
>>> # We can call rescale successfully without having Pillow
>>> rescale_img = rescale(img, 10)
>>> print(rescale_img.shape)
(3, 244, 360)

>>> # center_crop call raises error
>>> cropped_img = center_crop(img, (100, 100))
Traceback (most recent call last):
  File "/Users/amyroberts/code/transformers/../scripts/test_image_transforms_imports.py", line 10, in <module>
    cropped_img = center_crop(img, (100, 100))
  File "/Users/amyroberts/code/transformers/src/transformers/utils/import_utils.py", line 1073, in wrapper
    raise ImportError(f"Method `{func.__name__}` requires Pillow.")
ImportError: Method `center_crop` requires Pillow.
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/image_transforms.py [hunk 1]
```diff
@@ -115,10 +117,11 @@ def rescale(
     return rescaled_image
 
 
+@vision_required
```
Comment (sgugger) [inline src/transformers/image_transforms.py:120]: How is this different than having `requires_backend(["vision"])` as the first line of the function?

**Review by sgugger — COMMENTED** [review]
Thansk for working on this. Not too sur we actually need a decorator for that, with the existing tools.

File: src/transformers/image_transforms.py (hunk 1 above)
Comment (amyeroberts) [reply src/transformers/image_transforms.py:120]: To be honest, I just saw the `tf_required` and copied from there. I'll remove the decorator and add those instead 👍

**Review by sgugger — APPROVED** [review]
Perfect, thanks!

**Comment by ydshieh** [comment]
A quick question: If we add `requires_backends(center_crop, ["vision"])` in some methods in `src/transformers/image_transforms.py`, shouldn't we do the same in `src/transformers/image_utils.py`, say, `load_image`?

(And another minor question is, even if Pillow is installed, is it good to check `requires_backends(center_crop, ["vision"])` in each call? I feel for method like `load_image`, we can leave it as NameError, as it would be clearly the problem of Pillow itself.)

**Comment by amyeroberts** [comment]
> A quick question: If we add `requires_backends(center_crop, ["vision"])` in some methods in `src/transformers/image_transforms.py`, shouldn't we do the same in `src/transformers/image_utils.py`, say, `load_image`?

Yep. I'll add those in too.

**Review by ydshieh — APPROVED** [review]
Thank you @amyeroberts 

(I was think if it is good to call `require_backends` in each call - but it seems this call is very fast so there should be no problem)

File: src/transformers/image_transforms.py (hunk 1 above)
Comment (amyeroberts) [reply src/transformers/image_transforms.py]: Opened a draft PR here: https://github.com/huggingface/transformers/pull/20715

**Pull request merged**